src/firmware/app/control/wheel_controller.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "firmware/shared/circular_buffer.h"
+
+typedef struct WheelController WheelController_t;
+
+/**
+ * Function creates a WheelController_t with the specified parameters and returns a
+ * pointer to it. The wheel_controller_destroy function MUST be called before the
+ * WheelController_t* goes out of scope.
+ * The WheelController_t allows previously sampled wheel states to be added to it's state,
+ * and allows for new wheel commands to be requested.
+ *
+ * The coefficients parameters are in order of decreasing recentness - where
+ * coefficient_array[0] is the most recent value.
+ *
+ * The convention is as follows:
+ * Where 'z' is the discrete time operator and z^M is equivalent to M time-steps BACKWARDS
+ * in time. coeff[0] -> z^0 coeff[1] -> z^-1 coeff[2] -> z^-2
+ * ...
+ * coeff[N] -> z^-N
+ *
+ * NOTE: All coefficients MUST be specified. That is if the difference equation is A*z^-3
+ * + B*z^-1 the z^-2 coefficient MUST be included. Ex, A*z^-3 + 0*z^-2 + B*z^-1.
+ *
+ * The difference equation created is:
+ *      Output_voltage = Command_coefficients*Command_values -
+ * Output_sample_coefficients*output_values
+ *
+ * @param command_coefficients [in] The coefficients for the controller command inputs.
+ * These are the coefficients relating to how previous input effects the controller
+ * output.
+ *
+ * @param num_command_coefficients [in] The number of elements in the command_coefficients
+ * array
+ *
+ * @param output_sample_coefficients [in] The coefficients for the controller sampled
+ * output. These are the coefficients relating to how the sampled output(wheel state)
+ * effects the controller output.
+ *
+ * @param num_sample_coefficient [in] The number of elements in the
+ * sampled_output)coefficients array.
+ *
+ * @return Pointer to the newly created WheelController_t
+ */
+WheelController_t* app_wheel_controller_create(
+    float* command_coefficients, unsigned int num_command_coefficients,
+    float* output_sample_coefficients, unsigned int num_output_sample_coefficients);
+
+/**
+ * Function pushes a new command value to the specified WheelController. The command has
+ * the same units as the desired output. Ex. Command = 1m/s means that the user desires a
+ * wheel speed of 1m/s.
```

Comment:
How does this work?  I can provide any units and it will return values in the same units?

Does that mean it is technically unitless? Can we reduce the spec and limit it to SI units? (i.e m/s only)

Reply:
I've updated the comment to say '...*should* have the same units as the desired output"

If you request a speed of 1m/s you are expecting to get an output wheel speed in m/s essentially